src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -178,6 +188,60 @@ public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String d
         }
     }
 
+    public DRSAccessURL getAccessUrlForObjectId(AuthenticatedUserRequest authUser, String objectId, String accessId) {
+        DRSObject object = lookupObjectByDrsId(authUser, objectId, false);
+
+        DrsId drsId = drsIdService.fromObjectId(objectId);
+        Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
+        SnapshotProject snapshotProject = snapshotService.retrieveAvailableSnapshotProject(snapshot.getId());
+        GoogleProjectResource projectResource = snapshot.getProjectResource();
+
+        FSItem fsObject = null;
+        try {
+            String lookupTimer = performanceLogger.timerStart();
+            fsObject = fileService.lookupSnapshotFSItem(
+                snapshotProject,
+                drsId.getFsObjectId(),
+                1);
+
+            performanceLogger.timerEndAndLog(
+                lookupTimer,
+                objectId, // not a flight, so no job id
+                this.getClass().getName(),
+                "fileService.lookupSnapshotFSItem");
+        } catch (InterruptedException ex) {
+            throw new FileSystemExecutionException("Unexpected interruption during file system processing", ex);
+        }
+
+        GcsPdao.GcsLocator locator = GcsPdao.getGcsLocatorFromGsPath(fsObject.getPath());
+        String gsBucket = locator.getBucket();
+
+        if (doesAccessIdMatch(accessId, object)) {
+            Storage storage = StorageOptions.newBuilder().setProjectId(projectResource.getGoogleProjectId()).build().getService();
+
+            // Define resource
+            BlobInfo blobInfo = BlobInfo.newBuilder(BlobId.of(gsBucket, object.getName())).build();
+
+            URL url =
+                storage.signUrl(blobInfo, 15, TimeUnit.MINUTES, Storage.SignUrlOption.withV4Signature());
```

Comment:
Replacing the `15` with a variable describing what it is would be helpful here

Reply:
+1 I'd make this a constant and we may want to look into if there's anything in the spec about making that configurable